### ISSUE TITLE
Add token to metamask buttons

### ISSUE THREAD

**Issue by yellowgh0st** [vetherasset/vader-dapp#133]
- [ ] VADER
- [ ] xVADER

**Comment by Mervyn853** [comment]
Any updates on this for xVADER and VADER so users do not need to search for token contract address? @yellowgh0st

**Comment by yellowgh0st** [comment]
Not yet adding to 2do

**Comment by dimorphic** [comment]
#sneaky #wip ☯️👀

<img width="1590" alt="vader-tokens-wip" src="https://user-images.githubusercontent.com/3442665/148512213-74707ae4-404b-479b-97f6-015dc933661a.png">

**Comment by yellowgh0st** [comment]
@dimorphic are you on discord?

**Comment by dimorphic** [comment]
@yellowgh0st : Aye, sir! @NyX . I'll ping you laterz, got some IRL duties to attend 🤖

**Comment by yellowgh0st** [comment]
Instead of a separate page, what would you think about a modal with such content that will pop up after the user takes specific actions. In other words, a recommendation to add the token after stake/acquire/bond purchase.

**Comment by dimorphic** [comment]
I like the idea of `recommendation to add`, to enhance current / specific actions, a tooltip or something to remind the user.
But! I think the **only fitting place** to do this is in the staking tab & bond for now, more precise, 1st time staking Vader tokens. Because:

1. Users that *would come to stake* `V`, would be already tracking `V` in their wallets (where do they get the info from?)
2. If they do the approve & stake `V` steps, app should show them a recommendation to start tracking `xV` token, as most or even all won't be already tracking it prior staking, as not everyone *stakes*, knows of or even affords to do it up-front after acquiring `V`, they would rather get more `V` I guess and stake laterz. Still, they need the address to track in wallet...
3. In Bonds maybe we can do the same, even if I think that ppl who get bonds already are tracking `V`
4. In Acquire for sure, once that part is available

Otherwise, in the bond / acquire views, the user is most likely aware of both tokens and already interacted with the protocol in some way, thus most likely they would be already tracking both tokens, right? Am I missing something? :/

As I see it, users will look to add & track a custom token(s) into their wallets like so (abstract):
1. Before acquiring a token
2. After acquiring a token

So, considering numero #1 & #2 - most people will get Vader somehow, from somewhere (swap, CEX, etc) and have it delivered to their wallet. In this case, they don't interact / take a specific action in the Vader dApp that we can't track & act upon (currently no `Eth/Vader` swap in dApp besides Bonds kinda, but not rly). It doesn't apply for `V` token, but your suggestion sure does work for `xV` right after 1st time staking, to recommend it for add.

Still, it would be better to have the tokens listed somewhere right in the Vader dApp (as in the official links/sources) so the user can copy the contract address fast & add the custom token to his wallet and be done with it, just go 🚀 ; instead of opening some 3rd party site (in my case it was CMC) to lookup for the correct contract address. Should be official info in the dApp - visible and always reachable fast if you needz it, thus a page? #imo #ux Also, can we let them track the tokens without paying each time 300 🍌  in Eth gas fees to show a popup? 🤣  :D

As I said, I like the idea, but I think it's only fitting for the 1st time staking part (via local storage boolean), because sadly there's no way to check if some tokens are *being tracked or not* in a web3 wallet. :<

TL;DR;  : do both, kinda. ¯\_(ツ)_/¯

@yellowgh0st thoughts? Does it make any sense? (-.-)

**Comment by yellowgh0st** [comment]
Regards the recommendation cases you are right, my intention was only to briefly picture the idea.

> Still, it would be better to have the tokens listed somewhere right in the Vader dApp (as in the official links/sources) so the user can copy the contract address fast & add the custom token to his wallet and be done with it, just go rocket ; instead of opening some 3rd party site (in my case it was CMC) to lookup for the correct contract address. Should be official info in the dApp - visible and always reachable fast if you needz it, thus a page? #imo #ux Also, can we let them track the tokens without paying each time 300 banana in Eth gas fees to show a popup? rofl :D

Yes, you're right, it's good information to have yet it is still **secondary thing** and unnecessary for operation. That header menu should be exclusive to primary functions in order to keep it clean and easy.
The user definitely doesn't pay any gas for pop-up or to track tokens. I'm not sure what do you mean by that part. They only pay if they submit contract interaction which most of the stuff on the UI is not.

I have created wallet context menu where they can now import tokens. I recommend to think how in other ways this menu can be used to achieve what you describe to get the token information, like how we can design the menu item list etc. The left top area provides potential space for more useful indicators.

I have also an idea that every token symbol might be treated as a sort of interactive tag that might do something when's hit egg. copy the address into the clipboard or store the token object somewhere.

**Comment by dimorphic** [comment]
I agree the menu header should stay as clean as possible. 👍 I do like the wallet submenu you did, especially the `disconnect` button! It was needed and I wanted to add it also. :D The Eth gas was a joke regarding not having to interact with a contract to get the 'recommendation' popup. :) Anyways!

I already have a suggestion on how to combine these 2: keeping in mind that we wants clean header bar, I'd take the same approach with the wallet submenu too, keep it as clean as possible, bare minimum. It looks great on desktop, but on mobile devices, when in burger mode, currently it feels like i'm in some burger-menu-land-ception :D Like so (3 levels):

<img width="408" alt="Screenshot 2022-01-08 at 03 34 35" src="https://user-images.githubusercontent.com/3442665/148627865-cd4ebcac-f849-4e35-acc3-cf3334e303fd.png">

I know, there's so much you can do in that much/little space and so many items to lists. So: the `disconnect` has to be there for sure, but I would merge your 2x '+' buttons into a single one 'Track in wallet' (no confusion with the 'add into wallet'  - sounds less like deposit/funding this way maybe?) and have that route link redirect to the `/tokens` page view. This doesn't really solve the 'multi 3 level' problem in burger mode, but at least we keep the wallet submenu 'minimal' and navbar clear this way.

My proposal(s):

https://user-images.githubusercontent.com/3442665/148629656-79745c2e-3fc6-4508-bccc-9dde259b0201.mp4


https://user-images.githubusercontent.com/3442665/148629700-3428ca6f-3b4f-481f-be4b-5fad91689ed4.mp4

2x 🧠  = 🚀 go brrrrrr! :D

P.S.: bonus inc on Discord ☯️
